tests/qinfo/test_trace_distance.py

```diff
@@ -0,0 +1,543 @@
+        )((qml.numpy.array(param, requires_grad=True)))
+        expected_td = expected_grad_trace_distance_rx_pauliz(param)
+        assert qml.math.allclose(td_grad, expected_td)
+
+        td_grad = qml.grad(
+            qml.qinfo.trace_distance(circuit1, circuit0, wires0=[wire - 1], wires1=[wire - 1])
+        )(None, qml.numpy.array(param, requires_grad=True))
+        assert qml.math.allclose(td_grad, expected_td)
+
+    interfaces = ["auto", "autograd"]
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("param", parameters)
+    @pytest.mark.parametrize("wire", wires)
+    @pytest.mark.parametrize("interface", interfaces)
+    def test_trace_distance_qnodes_rx_tworx_grad(self, param, wire, interface):
+        """Test the gradient of the trace distance between two trainable circuits."""
+        dev = qml.device("default.qubit", wires=wire)
+
+        @qml.qnode(dev, interface=interface, diff_method="backprop")
+        def circuit(x):
+            qml.RX(x, wires=wire - 1)
+            return qml.state()
+
+        td_grad = qml.grad(
+            qml.qinfo.trace_distance(circuit, circuit, wires0=[wire - 1], wires1=[wire - 1])
+        )(
+            (qml.numpy.array(param, requires_grad=True)),
+            (qml.numpy.array(2 * param, requires_grad=True)),
+        )
+        expected = expected_grad_trace_distance_rx_pauliz(param)
+        expected_td = [-expected, expected]
+        assert qml.math.allclose(td_grad, expected_td)
+
+    interfaces_func = [
+        ("auto", lambda x: x),
+        ("torch", torch.tensor),
+        ("tf", tf.Variable),
+        ("jax", jax.numpy.array),
+        ("jax-jit", jax.numpy.array),
+    ]
+
+    @pytest.mark.all_interfaces
+    @pytest.mark.parametrize("device", devices)
+    @pytest.mark.parametrize("param", parameters)
+    @pytest.mark.parametrize("wire", wires)
+    @pytest.mark.parametrize("interface_func", interfaces_func)
+    def test_trace_distance_qnodes_rx_pauliz(self, device, param, wire, interface_func):
+        """Test the trace distance between Rx and PauliZ circuits with multiple interfaces."""
+        interface, func = interface_func
+
+        if device == "lightning.qubit" and interface == "jax-jit":
+            pytest.skip("Can't use lightning.qubit device with JAX JIT.")
+
+        dev = qml.device(device, wires=wire)
+
+        @qml.qnode(dev, interface=interface)
+        def circuit0(x):
+            qml.RX(x, wires=wire - 1)
+            return qml.state()
+
+        @qml.qnode(dev, interface=interface)
+        def circuit1():
+            qml.PauliZ(wires=wire - 1)
+            return qml.state()
+
+        td = qml.qinfo.trace_distance(circuit0, circuit1, wires0=[wire - 1], wires1=[wire - 1])(
+            func(param)
+        )
+        expected_td = expected_trace_distance_rx_pauliz(param)
+        assert qml.math.allclose(td, expected_td)
+
+        # TODO: Commented until #4194 is resolved
```

Comment:
Btw @eddddddy I've seen that you issued the #4205 PR in order to fix the associated issue. However, if I decomment this line, the Github Actions will fail. Should we wait for your PR to be merged to merge this one, or is it preferable that you uncomment this test within your PR?

Reply:
For now just leave this part commented.